playbooks/common/openshift-cluster/upgrades/v3_9/upgrade_control_plane.yml

```diff
@@ -17,32 +17,34 @@
     l_init_fact_hosts: "oo_masters_to_config:oo_etcd_to_config:oo_lb_to_config"
     l_base_packages_hosts: "oo_masters_to_config:oo_etcd_to_config:oo_lb_to_config"
 
-## Check to see if they're running 3.7 and if so upgrade them to 3.8 on control plan
-## If they've specified pkg_version or image_tag preserve that for later use
-- name: Configure the upgrade target for the common upgrade tasks 3.8
+- name: Configure the initial upgrade target for the common upgrade tasks
   hosts: oo_masters_to_config:oo_etcd_to_config:oo_lb_to_config
   tasks:
   - set_fact:
-      openshift_upgrade_target: '3.8'
+      # We use 3.9 here so when we run openshift_version we can get
+      # correct values for 3.9, 3.8 we will hard-code the values in
+      # ../pre/version_override.yml, if necessary.
+      openshift_upgrade_target: '3.9'
       openshift_upgrade_min: '3.7'
-      openshift_release: '3.8'
-      _requested_pkg_version: "{{ openshift_pkg_version if openshift_pkg_version is defined else omit }}"
-      openshift_pkg_version: ''
-      _requested_image_tag: "{{ openshift_image_tag if openshift_image_tag is defined else omit }}"
+
+## Check to see if we need to double upgrade (3.7 -> 3.8 -> 3.9)
+- name: Configure variables for double upgrade
+  hosts: oo_masters_to_config:oo_etcd_to_config
+  tasks:
+  - set_fact:
       l_double_upgrade_cp: True
+      l_version_override_hosts: "oo_masters_to_config:oo_etcd_to_config"
+      l_double_upgrade_first_version: "3.8.0"
+      l_double_upgrade_first_release: "3.8"
+      l_double_upgrade_first_tag: "v3.8.0"
```

Comment:
Same comment as above.  I don't see a reason not to do this.

Reply:
"v3.8.0" is not the latest 3.8 image tag.

Reply:
It will be the latest release image tag when we release the software: https://hub.docker.com/r/openshift/origin/tags/

If we want to go back and bump it to v3.8.1 after v3.8.1 is released, that would be fine.  Of course, a user could use extra_vars to override these components if it was absolutely necessary, or alternatively upgrade using 3.8's plays and then upgrade with 3.9's.  So, I think we're still providing a lot of flexibility here, just not in the traditional manner.